src/consumer/__tests__/consumeMessages.spec.js

```diff
@@ -205,6 +206,99 @@ describe('Consumer', () => {
     }
   })
 
+  it('after an error, aborts pending batch processing before retrying', async () => {
```

Comment:
Let's make sure that these tests are verifying what we think they're verifying. For this one, it means we expect eachBatch to have been called *once* for any partition above the concurrency limit, and *twice* for any partition within the concurrency limit.

Meaning if we have 3 partitions and a concurrency limit of 2, we expect eachBatch to have been called twice for partitions 1 and 2, and just once for partition 3 (because we skip it since there was a previous error).

Reply:
I re-worked the tests cases.  please take another look.